src/backend/session/direct.rs

```diff
@@ -403,9 +403,10 @@ impl BoundDirectSession {
     pub fn unbind(self) -> DirectSessionNotifier {
         let BoundDirectSession { source, notifier } = self;
         source.remove();
-        match Rc::try_unwrap(notifier) {
```

Comment:
Did clippy suggest this? I don't actually find the new code more readable than the old.

Reply:
Probably, because we are not using the Err(_)-contents. But it is fine in my eyes.

Reply:
Yeah, it's not like it's become clearly worse, but I'd say it's also not clearly better.

Reply:
Oh, it complained that you're `panic!`ing with a general message, ignoring what's inside the `Err`?

Reply:
Yes, it was hitting [`match_wild_err_arm`](https://rust-lang.github.io/rust-clippy/master/#match_wild_err_arm)

Reply:
Well, if is not like the `if let` is much better. The clippy warning would be justified, if we could do something else then ignoring the error or panic'ing here, but that is not the case.

So I don't care, if we keep the change or add an explicit `#[allow(...)]` for `match_will_err_arm` here. Whatever you like better @vberger.

Reply:
I guess a third alternative would be

```rust
.unwrap_or_else(|| panic!("....."))
```

Might be a little better?

I'm not really fan of sparkling a lot of `#[allow(...)]` everywhere tbqh.